google/datalab/bigquery/_query.py

```diff
@@ -33,49 +34,47 @@ class Query(object):
   This object can be used to execute SQL queries and retrieve results.
   """
 
-  def __init__(self, sql, env=None, udfs=None, data_sources=None,
-               subqueries=None):
+  def __init__(self, sql, env=None, udfs=None, data_sources=None, subqueries=None):
     """Initializes an instance of a Query object.
 
     Args:
       sql: the BigQuery SQL query string to execute
       env: a dictionary containing objects from the query execution context, used to get references
           to UDFs, subqueries, and external data sources referenced by the query
       udfs: list of UDFs referenced in the SQL.
-      data_sources: dictionary of external data sources referenced in the SQL.
+      data_sources: list of external data sources referenced in the SQL.
       subqueries: list of subqueries referenced in the SQL
 
     Raises:
       Exception if expansion of any variables failed.
       """
     self._sql = sql
-    self._data_sources = data_sources
     self._udfs = udfs
     self._subqueries = subqueries
     self._env = env or {}
+    self._data_sources = []
```

Comment:
Here we are setting `self._data_sources` to be a list, but below it is set to a dictionary.

We should be consistent and set it to an empty dictionary here (or None).